### PR TITLE
Studio:: Portal access restriction

### PR DESCRIPTION
**Pull request by medou-boushab**
The objective of this card is to enable portal access restrictions in the studio.

### CODE REVIEW

File: app/scripts/proactive/rest/studio-client.js [hunk 1]
```diff
@@ -140,25 +179,43 @@ define(
...
+                            if(response){
+                                $.ajax({
+                                    type: "GET",
+                                    url: config.restApiUrl + "/connected",
+                                    beforeSend: function(xhr) {
+                                        xhr.setRequestHeader('sessionid', localStorage['pa.session'])
+                                    },
+                                    success: function(data) {
+                                        if (data) {
+                                            console.log("Connected to the studio", data)
```
Comment (amouhoub) [inline app/scripts/proactive/rest/studio-client.js]: Please remove the logs

File: app/scripts/proactive/rest/studio-client.js [hunk 2]
```diff
@@ -140,25 +179,43 @@ define(
...
+                                            success()
+                                        } else {
+                                            console.log("Not connected to the studio", data)
```
Comment (amouhoub) [inline app/scripts/proactive/rest/studio-client.js]: please remove the logs

File: app/scripts/proactive/rest/studio-client.js [hunk 3]
```diff
@@ -85,9 +104,29 @@ define(
...
+                                beforeSend: function(xhr) {
+                                    xhr.setRequestHeader('sessionid', localStorage['pa.session'],
+                                                        'Content-Type', 'application/json')
+                                },
+                                success: function(response) {
+                                    if(response){
+                                        that.alert("Connected", "Successfully connected user", 'success');
+                                        return onSuccess();
+                                    } else {
+                                        that.alert("Cannot connect to ProActive Studio", 'The access to studio portal is not authorized' , 'error');
```
Comment (amouhoub) [inline app/scripts/proactive/rest/studio-client.js]: never use an alert in javascript, it's only for debugging

File: app/scripts/proactive/rest/studio-client.js [hunk 4]
```diff
@@ -85,9 +104,29 @@ define(
...
+                                    }
+
+                                },
+                                error: function(data) {
+                                    console.log("Failed to know the user permission", data)
```
Comment (amouhoub) [inline app/scripts/proactive/rest/studio-client.js]: console.log -> console.error

File: app/scripts/proactive/rest/studio-client.js [hunk 5]
```diff
@@ -140,25 +179,43 @@ define(
...
-                                fail()
+                                fail();
                             }
                         },
-                        error: function(data) {
-                            console.log("Not connected to the studio", data)
-                            localStorage.removeItem('pa.session')
-                            fail()
+                        error: function(response) {
+                            console.error("Failed to know the user permission", response)
```
Comment (mboussaa) [inline app/scripts/proactive/rest/studio-client.js]: User is not authorized

Comment (medou-boushab) [reply app/scripts/proactive/rest/studio-client.js]: It's the case where the request failed, we don't know if he is authorized or not

File: app/scripts/proactive/rest/studio-client.js (hunk 4 above)
Comment (medou-boushab) [reply app/scripts/proactive/rest/studio-client.js]: ok

File: app/scripts/proactive/rest/studio-client.js (hunk 1 above)
Comment (mboushab) [reply app/scripts/proactive/rest/studio-client.js]: done

File: app/scripts/proactive/rest/studio-client.js (hunk 5 above)
Comment (mboussaa) [reply app/scripts/proactive/rest/studio-client.js]: so: "Unknown User Permission"

**Comment by mboushab** [comment]
retest this please

File: app/scripts/proactive/rest/studio-client.js (hunk 5 above)
Comment (mboushab) [reply app/scripts/proactive/rest/studio-client.js]: ok

**Pull request merged**